File: khalti-android/src/main/java/com/khalti/checkOut/EBanking/deepLinkReceiver/DeepLinkPresenter.java
package com.khalti.checkOut.EBanking.deepLinkReceiver;

import androidx.annotation.NonNull;

import com.khalti.checkOut.helper.Config;
import com.khalti.checkOut.api.OnCheckOutListener;
import com.khalti.utils.EmptyUtil;
import com.khalti.utils.GuavaUtil;
import com.khalti.utils.Store;

import java.util.HashMap;

class DeepLinkPresenter implements DeepLinkContract.Listener {
    @NonNull
    private final DeepLinkContract.View mDeepLinkView;

    DeepLinkPresenter(@NonNull DeepLinkContract.View mDeepLinkView) {
        this.mDeepLinkView = GuavaUtil.checkNotNull(mDeepLinkView);
        mDeepLinkView.setListener(this);
    }

    @Override
    public void receiveEBankingData() {
        HashMap<String, Object> map = mDeepLinkView.receiveEBankingData();
        Config config = EmptyUtil.isNotNull(Store.getConfig()) ? Store.getConfig() : mDeepLinkView.getConfigFromFile();

        if (EmptyUtil.isNotNull(map) && EmptyUtil.isNotNull(config)) {
            OnCheckOutListener onCheckOutListener = config.getOnCheckOutListener();
            onCheckOutListener.onSuccess(map);
        }
        mDeepLinkView.closeDeepLink();
//        RxBus.getInstance().post("close_check_out", null);
        Store.getCheckoutEventListener().closeCheckout();
    }
}

File: khalti-android/src/main/java/com/khalti/checkOut/ebanking/deepLinkReceiver/DeepLinkPresenter.kt
package com.khalti.checkOut.ebanking.deepLinkReceiver

import com.khalti.utils.EmptyUtil
import com.khalti.utils.GuavaUtil
import java.util.*

internal class DeepLinkPresenter(view: DeepLinkContract.View) : DeepLinkContract.Presenter {

    private val view: DeepLinkContract.View = GuavaUtil.checkNotNull(view)

    init {
        view.setPresenter(this)
    }

    override fun onCreate() {
//        val map = view.receiveEBankingData()
        val map = object : HashMap<String, Any>() {
        }
//        val config = if (EmptyUtil.isNotNull(Store.getConfig())) Store.getConfig() else view.configFromFile

        val config = view.configFromService
        if (/*EmptyUtil.isNotNull(map) &&*/ EmptyUtil.isNotNull(config)) {
            val onCheckOutListener = config!!.onCheckOutListener
            onCheckOutListener.onSuccess(map!!)
        }
        view.closeDeepLink()
//        Store.getCheckoutEventListener().closeCheckout()
    }

    override fun onDestroy() {
    }
}